src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -152,7 +152,12 @@ bool emberAfGeneralCommissioningClusterSetRegulatoryConfigCallback(app::CommandH
                                                                    const Commands::SetRegulatoryConfig::DecodableType & commandData)
 {
     DeviceControlServer * server = &DeviceLayer::DeviceControlServer::DeviceControlSvr();
-    CheckSuccess(server->SetRegulatoryConfig(commandData.location, commandData.countryCode, commandData.breadcrumb), Failure);
+
+    static_assert(std::is_same<uint8_t, decltype(to_underlying(commandData.location))>::value,
```

Comment:
static_assert and static_cast seems to be repeated  ... a separate header method may be more  centralized. Eventually we could force SetRegulatoryConfig take in a class type to force the conversion method (i.e. a static_cast equivalent) to be used.